packages/react-pages/src/node/index.ts

```diff
@@ -43,55 +54,53 @@ export default function pluginFactory(
         },
       },
     }),
-    configResolved: (config) => {
-      if (!pagesDir) {
-        pagesDir = path.resolve(config.root, 'pages')
-      }
+    configResolved(config) {
+      pagesDir = opts.pagesDir ?? path.resolve(config.root, 'pages')
+      pageStrategy = new PageStrategy(pagesDir, findPages, loadPageData)
     },
-    resolveId(importee, importer) {
-      if (importee === '@!virtual-modules/pages') {
-        // page list
-        return importee
-      }
-      if (importee.startsWith('@!virtual-modules/pages/')) {
-        // one page data
-        return importee
-      }
-      if (importee === '@!virtual-modules/theme') {
-        return importee
-      }
-      if (importee === '@!virtual-modules/ssrData') {
-        return importee
-      }
+    configureServer({ watcher }) {
+      pageStrategy
+        .on('promise', () => watcher.emit('change', pagesModuleId))
+        .on('change', (pageId: string) =>
+          watcher.emit('change', pagesModuleId + pageId)
```

Comment:
Should it be `pagesModuleId + '/' + pageId`?

Reply:
`pageId` always starts with `/` :)